.github/workflows/self-tests-ubuntu.yaml

```diff
@@ -0,0 +1,31 @@
+name: Ubuntu self-tests
+
+on: push
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            install-script: sudo apt-get install make automake libtool libtool-bin autoconf doxygen gfortran g++
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install requirements
+      run: ${{ matrix.install-script }}
+    - name: Extra config
```

Comment:
For example, this stuff can be removed for the Ubuntu only workflow

Reply:
See https://github.com/oomph-lib/oomph-lib/pull/20#discussion_r678641150.

Reply:
**Update:** Removed the abstraction with a matrix and tidied the script up.